src/d3a/models/strategy/external_strategies/__init__.py

```diff
@@ -285,32 +285,60 @@ def _publish_trade_event(self, trade, is_bid_trade):
             # the bid clearing and one for the offer clearing.
             return
 
-        event_response_dict = {"device_info": self._device_info_dict,
-                               "event": "trade",
-                               "trade_id": trade.id,
-                               "time": trade.time.isoformat(),
-                               "price": trade.offer.price,
-                               "energy": trade.offer.energy,
-                               "fee_price": trade.fee_price,
-                               "area_uuid": self.device.uuid,
-                               "seller": trade.seller
-                               if trade.seller == self.device.name else "anonymous",
-                               "buyer": trade.buyer
-                               if trade.buyer == self.device.name else "anonymous",
-                               "residual_id": trade.residual.id
-                               if trade.residual is not None else "None"}
-
-        bid_offer_key = 'bid_id' if is_bid_trade else 'offer_id'
-        event_response_dict["event_type"] = "buy" \
-            if trade.buyer == self.device.name else "sell"
-        event_response_dict[bid_offer_key] = trade.offer.id
-
         if self.is_aggregator_controlled:
+            event_response_dict = {'event': 'trade',
```

Comment:
In general, it would be great instead of creating 2 long dicts, one for the aggregator and one for the device connection, that have some common key-value pairs, I would suggest to refactor this and create one dictionary for both cases, and then the aggregator/device can add the fields that are unique to them.

Reply:
Yeah, my motivation for this was that we will deprecate the direct sending of the trade event soonish.
Or do you think that this will be some time until then and stand by your comment?

Reply:
In this case I agree, feel free to leave it as is!